src/storage/validators/QuotaValidator.ts

```diff
@@ -27,6 +28,8 @@ export class QuotaValidator extends Validator {
 
     // 2. Check if the estimated size is bigger then the available size
     const estimatedSize = await this.strategy.estimateSize(metadata);
+    // Remove content-length property in metadata, as it is only ever used as input for estimating the size.
+    metadata.removeAll(CONTENT_LENGTH_TERM);
```

Comment:
@RubenVerborgh made the good comment that instead of hardcoding this in this specific class, it makes more sense to create a new small component that can filter out certain kinds of metadata, and then gets configured in the quota config to specifically remove the content-length metadata. That way we can reuse that component in similar situations in the future. I was thinking of some sort of `FilterMetadataDataAccessor` that takes as input some kind of triple patterns (in this case one that matches the content-length predicate) and removes the from the metadata before sending it to the `DataAccessor` it is wrapping around.

Reply:
Excellent idea! Will look into this asap!

Reply:
What is the best approach for modeling these triple patterns (within the Components.js config framework)? Are there currently other classes that take a similar construct as constructor input arguments?

Currently I'm experimenting with an array of the following type:

```
export interface MetadataFilter {
  subject?: string;
  predicate?: string;
  object?: string;
}
```

For which these optional string attributes should then be converted to `NamedNode`s. However, object could also be a literal, so I will need to add additional properties for object type and literal type (if the object is a literal)..

Reply:
As a reference @woutslabbinck has a similar interface in the metadata PR: https://github.com/CommunitySolidServer/CommunitySolidServer/pull/1188/files#diff-90a5ecb524cafe2925543514f4372e2cc7358fce416fb418c8c3b877ae5ec4d5R74-R84

Can't be reused though since it's not merged yet, so we might have to combine the solutions afterwards. But something like that indeed. It's a class there, but looking at it again it could indeed just be an interface, and then use `@range {json}` in the tsdoc so the components.js config stays clean.

Good point about the literal actually, that's something that I didn't think of. We could use a more specific serialization to take that into account, but my guess is that only named nodes will ever be relevant for this situation (this will be a fun comment to link to in the future when I'm wrong), so I would assume the input is always that.

Reply:
Thanks for the input! I'll start with supporting named nodes only for the object property.

Reply:
I'm having some issues with the config:

```
Could not build the config files from community-server/config/quota-file.json
Cause: Error while parsing file "community-server/config/storage/backend/quota/quota-file.json": Invalid predicate IRI: predicate
```

The type definitions are in `FilterMetadataDataAccessor.ts`. At first glance, I don't see what I'm doing different compared to Wouter's approach.

Reply:
Nevermind, I unintendedly used @id instead of @type in the config, silly mistake..